src/plugins/data/public/plugin.ts

```diff
@@ -20,16 +20,22 @@
 import { PluginInitializerContext, CoreSetup, CoreStart, Plugin } from '../../../core/public';
 import { AutocompleteProviderRegister } from './autocomplete_provider';
 import { DataPublicPluginSetup, DataPublicPluginStart } from './types';
+import { searchService } from './search';
+import { SearchService } from './search/search_service';
 import { getSuggestionsProvider } from './suggestions_provider';
 
 export class DataPublicPlugin implements Plugin<DataPublicPluginSetup, DataPublicPluginStart> {
   private readonly autocomplete = new AutocompleteProviderRegister();
+  private readonly searchService: SearchService;
 
-  constructor(initializerContext: PluginInitializerContext) {}
+  constructor(initializerContext: PluginInitializerContext) {
+    this.searchService = searchService(initializerContext);
```

Comment:
do we need to create this in the constructor or could it be postponed till `setup()` ?

Reply:
We could but then we'd need to store the initializerContext and make searchService possibly undefined.  Is there any benefit to postponing? It was my impression that constructor and setup are all run before Kibana loads up so it doesn't make a difference which it's in. If I'm wrong I can move it.